src/java/boa/compiler/visitors/TypeCheckingVisitor.java

```diff
@@ -1525,13 +1525,15 @@ protected boolean checkTupleArray(final List<BoaType> types) {
 		if (types == null)
 			return false;
 
-		final String type = types.get(0).toBoxedJavaType();
```

Comment:
What is this change?  It looks like you are undoing some code cleanup I did months ago?